src/client/opamClient.ml

```diff
@@ -1945,10 +1945,9 @@ let init
   gt, rt, default_compiler
 
 let check_installed ~build ~post t atoms =
-  let available = (Lazy.force t.available_packages) in
   let pkgs =
     OpamPackage.to_map
-      (OpamFormula.packages_of_atoms available atoms)
+      (OpamFormula.packages_of_atoms t.packages atoms)
```

Comment:
The performance improvement needs to be in its own PR.

Reply:
In fact, as the code is removed in the rest of the PR, shouldn't we not keep the commit?

Reply:
the code is moved elsewhere in the final commit and is here to offset the performance cost of looking at the dependencies recursively

Reply:
lgtm